Allow logging format configuration

ssbarnea opened the pull request:
Allow logging format configuration

This allows user to control log formatting strings.

Change-Id: I4c6aaddd1be241ec161be24571cf254d9217584d
Signed-off-by: Sorin Sbarnea <ssbarnea@redhat.com>

##### SUMMARY

This allows us to change the format of the log files which until now was hardcoded in ansible.

Relates to #13702 

##### ISSUE TYPE
 - Feature Pull Request

ssbarnea committed:
Allow logging format configuration

This allows user to control log formatting strings.

Change-Id: I4c6aaddd1be241ec161be24571cf254d9217584d
Signed-off-by: Sorin Sbarnea <ssbarnea@redhat.com>

bcoca commented:
FYI, i plan to push logging out of display and implement as a callback in the future.

ssbarnea committed:
Enable used of coloring styles

Allows us to fine tune log filtering per handler based on message type (style).

This is backward compatible as LogStyle objects
do evaluate as strings.

Change-Id: I6d3e913c97975d346dfa6ea21b130075862385bb
Signed-off-by: Sorin Sbarnea <ssbarnea@redhat.com>

ssbarnea commented:
@bcoca Any reasons for not merging this? I think that the propose interface would allow any kind of refactoring later.

mattclay commented:
In addition to the CI failures noted by the bot, the integration tests are also failing with a traceback:

```
2017-08-29 16:35:26 Traceback (most recent call last):
2017-08-29 16:35:26   File "/root/ansible/bin/ansible-playbook", line 130, in <module>
2017-08-29 16:35:26     display.error("Unexpected Exception, this is probably a bug: %s" % to_text(e), wrap_text=False)
2017-08-29 16:35:26   File "/root/ansible/lib/ansible/utils/display.py", line 270, in error
2017-08-29 16:35:26     self.display(new_msg, color=C.COLOR_ERROR, stderr=True)
2017-08-29 16:35:26   File "/root/ansible/lib/ansible/utils/display.py", line 114, in display
2017-08-29 16:35:26     msg = stringc(msg, color)
2017-08-29 16:35:26   File "/root/ansible/lib/ansible/utils/color.py", line 92, in stringc
2017-08-29 16:35:26     color_code = parsecolor(color)
2017-08-29 16:35:26   File "/root/ansible/lib/ansible/utils/color.py", line 75, in parsecolor
2017-08-29 16:35:26     r"|gray(?P<gray>[0-9]+)", color)
2017-08-29 16:35:26   File "/usr/local/lib/python2.7/re.py", line 141, in match
2017-08-29 16:35:26     return _compile(pattern, flags).match(string)
2017-08-29 16:35:26 TypeError: expected string or buffer
```

ssbarnea commented:
@bcoca Any updates regarding Ansible logging refactoring? The pressure for being able to have verbose logs inside files and brief console output becomes bigger and bigger. 

We need to find a way to save detailed (like -vv/-vvv) logs while keeping minimal console output. If console output is not enough user may look inside the detailed log file.

bcoca commented:
The logging keeps being de-prioritized, but some in core (not all) feel like we should stall changes until that plan is implemented, I suggest you bring this up in irc meeting so we can give you a final decision on this PR.

ssbarnea commented:
@bcoca Closing this. Ping me if you need some help or feedback on the logging refactoring as I have a strong interesting in it and important use cases: when logging may be captured by QA tools calling ansible (ansible-lint, molecule, ansible-test).

bcoca commented:
I don't think we'll have time for 2.11 but i def want to get it onto roadmap for 2.12, refactoring of Display class, callbacks and logging (mostly trying to design it so the former 2 use the latter).
